docs/source/development/documentation.md

```diff
@@ -0,0 +1,110 @@
+* The `development` directory contains details on code development for `pyrealm`.
+* The `api` directory contains some simple stub files that are used to link to API
+  content generated from docstrings.
+
+We use the `sphinx_external_toc` package to maintain a table of contents for the
+package: you will need to add new pages to `_toc.yml` for them to appear in the table.
+
+### MyST Markdown
+
+All of documentation in `docs/source` uses [MyST
+Markdown](https://myst-parser.readthedocs.io/en/latest/) rather than the
+reStructuredText (`.rst`) format. Markdown is easier to write and read and the MyST
+Markdown extension is a literate programming format that allows Markdown pages to be run
+using Jupyter to generate dynamic content to show package use.
+
+We only include markdown format files (`.md`) in the repository and not the standard
+Python Notebook (`.ipynb`) format. Although this means that the output of code cells
+does not display on GitHub, the JSON format `.ipynb` files are much bulkier and also
+change whenever the notebook is run, not just when there are meaningful changes to the
+content.
+
+When the package documentation is built in `sphinx` the `myst_nb` extension
+automatically runs any code in MyST markdown documents and renders the resulting code
+and outputs into the HTLM. The documentation can therefore contain dynamically generated
+examples of running code and output plots to demonstrate how to use the
+package. This build process also ensures that code in documentation is kept up to date
+with the package. The syntax for including code blocks in documents is shown below -
+this example includes a tag that causes the code to be concealed in the built
+documentation, but with a toggle button to allow it to be expanded.
+
+````{code-block}
+```{code-cell} python
+:tags: [hide-input]
+# This is just an example code cell to demonstrate how code is included in
+# the pyrealm documentation.
+```
+````
+
+### Docstrings
+
+The `pyrealm` package uses docstrings written in the [Google
+style](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html).
+This allows the function documentation to be stored alongside the code and it is included
+in the documentation using the `sphinx` `autodoc` extension. See the code itself for
+examples of the documentation formatting and typical content.
+
+Unlike the files in `docs/sources` pages, the docstrings in code need to be written
+using reStructuredText: at present the `autodoc` functions in `sphinx` currently rely on
+`rst` inputs.
+
+### Referencing
+
+Both the `docs/source` and docstrings uses the `sphinxcontrib-bibtex` package to support
+citations. This uses Latex like citation keys in the documentation to insert references
+and build a bibliography. The `sphinx` configuration in `docs/source/conf.py` provides a
+custom Author/Year citation style. The reference library in `source/refs.bib` needs to
+be kept up to date with the literature for the project.
+
```

Comment:
Could you please provide an example here, this was a bit of an issue with VE at the start.